src/generators/Generator.js

```diff
@@ -260,14 +260,26 @@ export default class Generator {
 					// export is last property, we can just return it
 					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `return ` );
 				} else {
-					// TODO ensure `template` isn't already declared
-					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `var template = ` );
+					// we need to avoid a conflict with anything on the top-level scope of the component
+					// however, determining which things these are is tricky, so we instead just avoid conflicts with any identifiers anywhere
+					const identifiers = new Set();
+					walk( js, {
+						enter ( node ) {
+							if ( node.type === 'Identifier' ) {
```

Comment:
Was this an alternate solution or something to be done along with `annotateWithScopes`? It looks like to me that we can do what we need to do without this.